From 5874f9b1ced9c29d7d590d95e254b252f657a160 Mon Sep 17 00:00:00 2001
From: rofl0r <rofl0r@users.noreply.github.com>
Date: Sun, 20 Jun 2021 00:34:08 +0100
Subject: [PATCH] GNUmakefile: explicitly add required CPPFLAGS for
 tic/compile.o

for unknown reasons this seems to be necessary in some setups.
with my GNU make 3.82, the rule for libterminfo/compile.o is applied
to tic/compile.o too, as it seems to detect that tic/compile.c is a symlink
to libterminfo/compile.c and therefore uses the rules for the file in
libterminfo/.

closes #48
---
 GNUmakefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/GNUmakefile b/GNUmakefile
index 25e5c2c..d302ce1 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -62,6 +62,8 @@ TIC_HOBJS=$(TIC_SRCS:.c=.ho)
 TOOL_TIC=$(TOOL_HOSTTIC)
 TIC_MAN=$(sort $(wildcard tic/*.1))
 
+tic/compile.o: CPPFLAGS+= -DTERMINFO_COMPILE -DTERMINFO_DB -DTERMINFO_COMPAT
+
 TI_SRCS= libterminfo/term.c libterminfo/ti.c libterminfo/setupterm.c \
          libterminfo/curterm.c libterminfo/tparm.c libterminfo/tputs.c
 TI_SRCS+=libterminfo/compile.c libterminfo/hash.c
